src/Actions/Search.js

```diff
@@ -4,22 +4,26 @@ import { push } from 'react-router-redux'
 const prefix = 'SEARCH'
 
 export const query = function(message) {
-  return function(dispatch, getState) {
-    let provider = getState().web3.provider
+  return async function(dispatch, getState) {
+    // do nothing if user submits empty search string
+    if (message === '') return
+
+    let web3Instance = getState().web3.web3Instance
     // This will request the block by either its has or number
-    web3Request("getBlock", [message, true], provider, (err, block) => {
-      if (block) {
-        dispatch(push(`/blocks/${block.number}`))
-        return
-      }
+    try {
+      let block = await web3Request('getBlock', [message, true], web3Instance)
+      dispatch(push(`/blocks/${block.number}`))
+      return
+    } catch(err) {
+      console.log('Block search error: ', err)
```

Comment:
Not sure if I should've left these in or not - will make future debugging easier, but will add noise to the debugging console on every search for a transaction hash.

Future improvments which are out of scope for this change: might've been better to check whether or not the search query was prefixed with `0x` here rather than try and fail on every attempt to search for a transaction. Can add a query length filter to determine whether it's a block hash or a tx hash in the future as well.

Reply:
Responding to this: 
> might've been better to check whether or not the search query was prefixed with 0x here rather than try and fail on every attempt to search for a transaction. 

Transactions and block hashes have the same length, and both start with `0x`. In the current implementation you can search with a `0x`-prefixed string and get the appropriate block or transaction, whichever matches. Side note: because `getBlock` also allows numbers, you can search by block number as well.

Reply:
Also, was the console lot intentionally left in? It's not an error insomuch as the search query didn't hit on the first type of thing searched.

Reply:
> Transactions and block hashes have the same length, and both start with 0x

I wasn't thinking about it too hard when I wrote that, and we don't even handle searching for block hashes yet. Either way, I think my point remains that we can at least determine whether or not something is a block number.

Reply:
> Was referring specifically to the console.log line.

Ah, 👍 

> we don't even handle searching for block hashes yet

We do handle searching by block hashes.  `getBlock` accepts a block hash, hence the current search structure.

Reply:
Oh, right! I should've realized!